_docs/reference/config/kubernetes-proxy-injection.md

```diff
@@ -0,0 +1,214 @@
+sidecar into a pod: manually using istioctl `kube-inject` or automatically 
+using the Istio sidecar injector webhook. 
+
+Manual injection modifies the controller configuration, e.g. deployment. It 
+does this by modifying the pod template spec such that *all* pods for that
+deployment are created with the injected sidecar. Adding/Updating/Removing
+the sidecar requires modifying the entire deployment.
+
+Automatic injection injects at pod creation time. The controller resource is 
+unmodified. Sidecars can be selectively added/updated/removed by (re)starting 
+pods after modifying the injection configuration.
+
+Manual and automatic injection use the same templated configuration. See TBD 
+for template configuration format.
+
+# Manual injection
+
+`kube-inject` is designed to be run offline without access to a running Kubernetes
+cluster. 
+
+Download a copy of the injection and mesh configmap. This might typically be retrieved 
+from a SCM as part of a CI/CD pipeline.
+
+```
+$ kubectl -n istio-system get configmap istio-inject -o=jsonpath='{.data.config}' > inject-config.yaml 
+$ kubectl -n istio-system get configmap istio -o=jsonpath='{.data.mesh}' > mesh-config.yaml
+```
+
+Run `kube-inject` over the input YAML file and save and deploy the injected YAML file.
+
+```
+$ istioctl kube-inject \
+    --injectConfigFile inject-config.yaml \
+    --meshConfigFile mesh-config.yaml \
+    --filename samples/sleep/sleep.yaml \
+    --output sleep-injected.yaml
+$ kubectl apply -f sleep-injected.yaml    
+```
+
+Alternatively, this can be performed in a single step. This uses the built-in default template and dynamically fetches the mesh configuration from the `istio` ConfigMap. Additional parameter overrides are available via flags (see `istioctl kube-inject --help`).
+
+```
+kubectl apply -f <(~istioctl kube-inject --meshConfigMapName=istio -f samples/sleep/sleep.yaml)
+```
+
+Verify that the sidecar has been injected into the deployment.
+
+```
+$ kubectl get deployment sleep -o wide
+NAME      DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE       CONTAINERS          IMAGES                             SELECTOR
+sleep     1         1         1            1           2h        sleep,istio-proxy   tutum/curl,unknown/proxy:unknown   app=sleep
+```
+
+# Automatic injection with mutating webhooks
+
+See https://kubernetes.io/docs/admin/admission-controllers/#validatingadmissionwebhook-alpha-in-18-beta-in-19 and https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.9/#mutatingwebhookconfiguration-v1beta1-admissionregistration for overview of webhook admission controller.
+
+## Prerequisites
+
+Kubernetes 1.9 cluster is required with `admissionregistration.k8s.io/v1beta1` enabled.
+
+```
+$ kubectl api-versions | grep admissionregistration.k8s.io/v1beta1
+admissionregistration.k8s.io/v1beta1
+```
+
+### GKE
+
+```bash
+gcloud container clusters create <cluster-name> \
+    --enable-kubernetes-alpha 
```

Comment:
Is admission registration not enabled by default since it is beta in k8s?

Reply:
1.9.1 is available for non-whitelisted early access users with alpha clusters (see https://cloud.google.com/kubernetes-engine/release-notes#january-16-2018).